packages/components/src/RadioGroup/RadioGroup.test.tsx

```diff
@@ -53,28 +40,76 @@ test("it should be able to disable options", () => {
 test("it should have unique ids on all radio options", () => {
   const { container, getAllByLabelText } = render(<MockRadioGroup />);
   const labels = getAllByLabelText("Two");
-  const radio1 = container.querySelector(`input#${labels[0].id}`);
-  const radio2 = container.querySelector(`input#${labels[1].id}`);
+  const radio1 = container.querySelector(
+    `input#${labels[0].id}`,
+  ) as HTMLInputElement;
+  const radio2 = container.querySelector(
+    `input#${labels[1].id}`,
+  ) as HTMLInputElement;
   expect(radio1.checked).toBeFalsy();
   expect(radio2.checked).toBeFalsy();
   fireEvent.click(labels[1]);
   expect(radio1.checked).toBeFalsy();
   expect(radio2.checked).toBeTruthy();
 });
 
-function MockRadioGroup() {
-  const [value, setValue] = useState("One");
-  const [valueTwo, setValueTwo] = useState("One");
+test("it should render an option from `label` prop", () => {
+  const { getByText } = render(
+    <RadioGroup value="" onChange={jest.fn()}>
+      <RadioOption value="foo" label="Radio" />
+    </RadioGroup>,
+  );
+
+  expect(getByText("Radio")).toBeInstanceOf(HTMLLabelElement);
+});
+
+test("it should render an option from `children` prop", () => {
+  const { getByText } = render(
+    <RadioGroup value="" onChange={jest.fn()}>
+      <RadioOption value="foo">Radio</RadioOption>
+    </RadioGroup>,
+  );
+
+  expect(getByText("Radio")).toBeInstanceOf(HTMLLabelElement);
+});
+
+test("it should render a description", () => {
+  const { getByText } = render(
+    <RadioGroup value="" onChange={jest.fn()}>
+      <RadioOption value="foo" description="A sound box" label="Radio" />
+    </RadioGroup>,
+  );
+
+  expect(getByText("A sound box")).toBeInstanceOf(HTMLParagraphElement);
+});
+
+interface MockProps {
+  onChange?(val: string): void;
+}
+
+function MockRadioGroup({ onChange }: MockProps) {
+  const [value, setValue] = useState("one");
+  const [valueTwo, setValueTwo] = useState("one");
   return (
     <>
-      <RadioGroup onChange={setValue} value={value}>
-        <RadioOption value="one">One</RadioOption>
-        <RadioOption value="two">Two</RadioOption>
+      <RadioGroup onChange={handleFirstChange} value={value}>
+        <RadioOption value="one" label="One" />
+        <RadioOption value="two" label="Two" />
       </RadioGroup>
-      <RadioGroup onChange={setValueTwo} value={valueTwo}>
-        <RadioOption value="one">One</RadioOption>
-        <RadioOption value="two">Two</RadioOption>
+      <RadioGroup onChange={handleSecondValue} value={valueTwo}>
+        <RadioOption value="one" label="One" />
+        <RadioOption value="two" label="Two" />
       </RadioGroup>
     </>
   );
+
+  function handleFirstChange(val: string) {
```

Comment:
This and the function below use two words `change` and `value` when I think they are referring to the same thing. Consider making the use the same word.